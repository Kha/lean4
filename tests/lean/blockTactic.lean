example (p : Prop) (h : True → True → p) : p := by
  apply h
  - trivial
  - trivial

example (p : Prop) (h : True → True → p) : p := by
  - apply h
    - trivial
    - trivial

example (p : Prop) (h : True → True → p) : p := by
  - apply h  -- should fail
  - trivial
  - trivial

example (p : Prop) (h : True → True → p) : p := by
  apply h
  - apply id
    trivial
  - apply id
    trivial

example (p : Prop) (h : False → False → p) : p := by
  apply h
  - trivial  -- should
  - trivial  -- both fail
