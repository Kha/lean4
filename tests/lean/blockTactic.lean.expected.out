blockTactic.lean:12:2-12:3: error: unsolved goals
case a
p : Prop
h : True → True → p
⊢ True

case a
p : Prop
h : True → True → p
⊢ True
blockTactic.lean:13:2-13:11: error: no goals to be solved
blockTactic.lean:25:4-25:11: error: tactic 'assumption' failed, 
case a
p : Prop
h : False → False → p
⊢ False
blockTactic.lean:26:4-26:11: error: tactic 'assumption' failed, 
case a
p : Prop
h : False → False → p
⊢ False
